README.md

```diff
@@ -111,6 +111,66 @@ When running on cluster node, you will also need to distribute this keytab, belo
+
+Known Limitations (Design decision)
+====
+
+> SAC only supports SQL/DataFrame API (in other words, SAC doesn't support RDD).
+
+SAC relies on query listener to retrieve query and examine the impacts. We may not investigate on supporting RDD for future unless strongly necessary.
+
+> All "inputs" and "outputs" in multiple queries are accumulated into single "spark_process" entity when there're multple queries running in single Spark session.
+
+Unfortunately, we got lost on history for this. We assume this approach could concentrate overall impacts on the single Spark process for who ran for when, but it's not pretty clear why application id was taken.
+
+We've filed #261 to investigate changing the unit of "spark_process" entity to query, but it doesn't mean we will change it soon. It will be address only if we see the clear benefits on changing it.
+
+> Only part of inputs are tracked in Streaming query.
+
+This is from design choice on "trade-off": Kafka source supports subscribing with "pattern" which SAC cannot enumerate all matching existing topics, or even all possible topics (even it's possible it doesn't make sense, though). 
+
+We found "executed plan" provides actual topics which each (micro) batch read and processed, but as a result, only inputs which participate on (micro) batch are being included as "inputs" in "spark_process" entity. 
+
+If your query are running long enough that it ingests data from all topics, it would have all topics in "spark_process" entity.
+
+> SAC doesn't support tracking changes on columns (Spark models).
+
+We found difficulty to make column entity be consistently "up-to-date" supposing multiple spark applications are running and each SAC in spark application are trying to update on Atlas. Due to the difficulty, we dropped supporting column tracking.
+
+We haven't found difficulty on table level, but once we collect difficulties on making table entity being consistent as well, we might also consider drop supporting DDL and blindly create the entities whenever they're referenced.
+
+This doesn't apply to Hive models, which central remote HMS takes care of DDLs and Hive Hook will take care of updates.
+
+> SAC doesn't track dropping tables (Spark models).
+
+"drop table" event from Spark only provides db and table name, which is NOT sufficient to create qualifierName - especially we separate two types of tables - spark and hive. 
+
+SAC depends on reading the Spark Catalog to get table information but Spark might already dropped the table when SAC notices the table is dropped, so that is NOT consistent. Due to this technical limitation, we have to drop this feature.
+
+> ML entities/events may not be tracked properly.
+
+We are concentrating on making basic features be stable: we are not including ML features on the target of basic features as of now. We will revisit once we are sure to resolve most of issues on basic features.
```

Comment:
Do users need to be aware of these details? Just state the current limitation and the proposed patch and the tracking will be improved post Spark 3.0.

Reply:
I intended to add details - this is not only for end users but also for any contributors. Given this is open source, ideally all materials should be here to avoid centralization of knowledge. If this would give confusion, I'll have separate doc to explain the details and leave only overview here.